src/octoprint/util/comm.py

```diff
@@ -2438,30 +2439,54 @@ def convert_line(line):
                 line, lower_line = convert_line(line)
 
                 ##~~ SD file list
+                longname = None
+                size = None
+                timestamp = None
                 # if we are currently receiving an sd file list, each line is just a filename, so just read it and abort processing
                 if self._sdEnabled and self._sdFileList and "End file list" not in line:
                     preprocessed_line = line
-                    fileinfo = preprocessed_line.split(None, 2)
-                    if len(fileinfo) == 3:
-                        # name, size, long name
-                        filename, size, longname = fileinfo
+                    fileinfo = preprocessed_line.split(None, 3)
```

Comment:
All of this is starting to get a complexity that it really should be moved into a dedicated parser function that has a ton of unit test coverage. I can do that after I merge this, or alternatively you look into it as part of this PR. What's your preference?

Reply:
Please do. I'm occasional programmer and unfamiliar with pytest (hope the tests I added are ok).